coders/composio_coders/swe.py

```diff
@@ -0,0 +1,227 @@
+    logger.setLevel(logging.DEBUG)
+    logger.addHandler(handler)
+    logger.propagate = False
+    return logger
+
+
+logger = setup_logger()
+
+
+class IssueConfig(BaseModel):
+    issue_id: str = Field(..., description="git issue id that agent is solving")
+    base_commit_id: str = Field(..., description="base commit id for which issue needs to be fixed")
+    issue_description: str = Field(..., description="description of the issue")
+
+
+class CoderAgentArgs(BaseModel):
+    repo_name: str = Field(..., description="repo name in which agent has to work")
+    agent_output_dir: str = Field(..., description="task output directory for storing agent-chat logs,"
+                                                   " task-logs, testbed etc")
+    agent_backstory_tmpl: str = Field(default=AGENT_BACKSTORY_TMPL,
+                                      description="backstory template for the agent to work on")
+    issue_description_tmpl: str = Field(default=ISSUE_DESC_TMPL)
+    issue_config: IssueConfig = Field(..., description="issue config, with issue description, repo-name")
+
+    def __init__(self, **data):
+        super().__init__(**data)
+        self.issue_config = IssueConfig(**data.get("issue_config", {}))  # Initialize IssueConfig here
+
+
+class CoderAgent:
+    def __init__(self, args: CoderAgentArgs):
+        # initialize logs and history logs path
+        self.args = args
+        self.model_env = {}
+        self.issue_config = args.issue_config
+        self.repo_name = args.repo_name
+        if not self.issue_config.issue_id:
+            raise ValueError("no git-issue configuration is found")
+
+        # initialize composio toolset
+        tool_set = ComposioToolSet()
+        self.composio_toolset = tool_set.get_tools(apps=[App.LOCALWORKSPACE,
+                                                         App.CMDMANAGERTOOL,
+                                                         App.HISTORYKEEPER,])
+        # initialize agent-related different prompts
+        self.agent_role = "You are the best programmer. You think carefully and step by step take action."
+        self.agent_goal = "Help fix the given issue / bug in the code. And make sure you get it working."
+        self.expected_output = "A patch should be generated which fixes the given issue"
+        self.agent_backstory_tmpl = args.agent_backstory_tmpl
+        self.issue_description_tmpl = args.issue_description_tmpl
+        # initialize logger
+        self.logger = logger
+        # initialize agent logs and history dict
+        self.task_output_logs = ""
+        self.agent_logs = {}
+        self.agent_history = {}
+        self.current_logs = []
+        self.setup_logs_path(args.agent_output_dir)
+
+    def setup_logs_path(self, agent_output_dir_path):
+
+        task_output_dir = script_dir / Path(agent_output_dir_path) / Path(
+            LOGS_DIR_NAME_PREFIX + "_" + datetime.datetime.now().strftime("%Y-%m-%d_%H-%M-%S"))
```

Comment:
This is supposed to be a python package, don't use local/development directories (`script_dir`) as reference for storing/caching any data. On the user's machine this path will be resolved to the `site-packages/composio_coder` which is not ideal. Either use a local caching directory where the command is invoked or use a global one like `~/.composio`

Reply:
```
composio-coder solve
ℹ️ Starting issue solving with the following configuration: {
    "repo_name": "some_name",
    "issue_id": "169",
    "base_commit_id": "",
    "issue_description": "Local install does not work"
}
INFO     Model environment configuration file not found.                                                                                                    
INFO     Model environment configuration file not found. 
Traceback (most recent call last):
  File "/Users/viraj/.local/share/virtualenvs/composio-txr_aOhK/bin/composio-coder", line 33, in <module>
    sys.exit(load_entry_point('composio-coder', 'console_scripts', 'composio-coder')())
  File "/Users/viraj/.local/share/virtualenvs/composio-txr_aOhK/lib/python3.10/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
  File "/Users/viraj/.local/share/virtualenvs/composio-txr_aOhK/lib/python3.10/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
  File "/Users/viraj/.local/share/virtualenvs/composio-txr_aOhK/lib/python3.10/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/viraj/.local/share/virtualenvs/composio-txr_aOhK/lib/python3.10/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/viraj/.local/share/virtualenvs/composio-txr_aOhK/lib/python3.10/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
  File "/Users/viraj/workspace/composio/coders/composio_coders/cli.py", line 101, in solve
    coder_agent.run()
  File "/Users/viraj/workspace/composio/coders/composio_coders/swe.py", line 180, in run
    llm = self.get_llm()
  File "/Users/viraj/workspace/composio/coders/composio_coders/swe.py", line 177, in get_llm
    raise ValueError(f"Invalid model environment: {self.model_env}")
ValueError: Invalid model environment: {}
```

Reply:
fixed